pkg/api/enums.go

```diff
@@ -0,0 +1,83 @@
+package api
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import "fmt"
+
+// OutboundType represents a routing strategy to provide egress to the Internet.
+type OutboundType int
+
+const (
+	OutboundTypeLoadBalancer OutboundType = iota
+
+	outboundTypeLength // private, must be last
+)
+
```

Comment:
if this is api-version specific, does this belong here?

Reply:
There are equivalent constants under `v20240610preview` but that brings up a good point that we may want to use a map for conversions.

Reply:
Gonna defer this to a subsequent PR.